### PR TITLE
fix:image not display

### LINKED ISSUE #89: Can't share images in embedded chat
### Description:

Shared images in embedded chat can't be opened. The uploaded files point to localhost.

### Steps to reproduce:

1. Upload a screenshot in the chat
2. Click on the link

### Expected behavior:

The image opens in a new tab

### Actual behavior:

image

### LINKED ISSUE #96: image not visible in chat on file upload
### Description:
The image is not visible on the chat screen, it is visible as a link instead of an image.
The image uploaded from the Embedded chat is coming as a file.

### Steps to reproduce:

1. Click on + icon to add image 
2. Hit enter to send the image
3. The image would come as a file in the application
4. That image would also come as a file in the rocket chat server

### Expected behaviour:
image

### Actual behaviour:

image

### PR DESCRIPTION
**Pull request by VenomFate-619**
# On uploading the image, it was not visible on chat as an image.
On uploading the image it was not visible on chat as an image.
After this PR it will be visible as an image instead of a file
## Acceptance Criteria fulfilment

- [x] The image is coming as an image on chat instead of a file
- [x] another file format can be uploaded too
- [x] Handling if `image_dimensions` are not present in the response

Fixes #96 
Fixes #89

## Video/Screenshots
image

https://user-images.githubusercontent.com/67755128/213980859-c3681d29-f36a-4ef3-91a6-7e9d9f551b6a.mp4

### CODE REVIEW

**Comment by VenomFate-619** [comment]
I made changes in `imageAttachement.js` to handle if `image_dimensions` is not present.
There are some response which lack image_demensions

`{
  "_id": "bmKuc98zZFf4ggMPm",
  "rid": "GENERAL",
  "ts": "2023-01-22T16:13:59.304Z",
  "file": {
    "_id": "WK3nDLmFnGLr8rWzL",
    "name": "file_example_WEBP_50kB.webp",
    "type": "image/webp"
  },
  "files": [
    {
      "_id": "WK3nDLmFnGLr8rWzL",
      "name": "file_example_WEBP_50kB.webp",
      "type": "image/webp"
    }
  ],
  "attachments": [
    {
      "ts": "1970-01-01T00:00:00.000Z",
      "title": "file_example_WEBP_50kB.webp",
      "title_link": "/file-upload/WK3nDLmFnGLr8rWzL/file_example_WEBP_50kB.webp",
      "title_link_download": true,
      "image_preview": "/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAVACADASIAAhEBAxEB/8QAGQAAAwADAAAAAAAAAAAAAAAAAAUGAQMH/8QAKBAAAgEEAAUCBwAAAAAAAAAAAQMCAAQFBgcREiFRMUEWMjNCYXHB/8QAGQEAAQUAAAAAAAAAAAAAAAAAAwABAgUG/8QAHBEAAQUBAQEAAAAAAAAAAAAAAQACAwQhERIi/9oADAMBAAIRAxEAPwC24mLstRtbO4hjoTsZzEXSP2g1Xa0MdDEqZh4iNq4CY5HzSPfor2fGuxzGgoYO8T7Us02z+H8Suyt3PcpXpGXf+1nI2xuHwrqeSRjR7VhsV6bLB3jl/Uis9P7rl3Bu7bldmuH35PUoEnq81SbBLI5yxbZJRNImfmJrdomgMxq2yu3SjKR584nuaVmoCz0NKhVtAgteedUnBzHdM5TkCfBp3j7pyYgwYe1FFAjOcRbGnU2ResuInr9R786w3JXSFErbIcvzRRTuJBxDiAJ1f//Z",
      "image_url": "/file-upload/WK3nDLmFnGLr8rWzL/file_example_WEBP_50kB.webp",
      "image_type": "image/webp",
      "image_size": 50408,
      "type": "file"
    }
  ],
  "msg": "",
  "groupable": false,
  "u": {
    "_id": "vmtwrF9F5urxJMxjp",
    "username": "amanali",
    "name": "Syed Aman Ali"
  },
  "_updatedAt": "2023-01-22T16:13:59.343Z",
  "urls": [],
  "mentions": [],
  "channels": []
}`

**Comment by VenomFate-619** [comment]
I think we change the hard-coded host from image src
`src={http://localhost:3000${attachment.image_url}}`
it should be `src={host${attachment.image_url}}`

Should I add `host` in userStore so that we access it anywhere, we can  set the host in the index.js file

File: src/components/Attachments/ImageAttachment.js [hunk 1]
```diff
@@ -2,18 +2,17 @@ import { Box } from '@rocket.chat/fuselage';
...
-        width={attachment.image_dimensions.width}
-      />
-    </Box>
-  );
-};
+const ImageAttachment = ({ attachment }) => (
+  <Box>
+    <p>{attachment?.description}</p>
+    <img
+      src={`http://localhost:3000${attachment.image_url}`}
```
Comment (sidmohanty11) [inline src/components/Attachments/ImageAttachment.js:9]: Awesome work! Can we please update all these hard-coded strings. We can create a separate function called getHost inside lib/api.js RocketChat instance where we return the host.

We can get host as props for all the attachment components from the parent attachment component

File: src/components/Attachments/ImageAttachment.js [hunk 2]
```diff
@@ -2,18 +2,17 @@ import { Box } from '@rocket.chat/fuselage';
...
+      height={attachment?.image_dimensions?.height ?? 357}
+      width={attachment?.image_dimensions?.width ?? 476}
```
Comment (sidmohanty11) [inline src/components/Attachments/ImageAttachment.js:11]: Is it responsive while we use it width let's say less than this?

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Great work @VenomFate-619 !

File: src/components/Attachments/ImageAttachment.js (hunk 1 above)
Comment (VipinDevelops) [reply src/components/Attachments/ImageAttachment.js:9]: Great work 👏
Sidharth bhaiya  i also need that get host function in one of my PR 

So should i make that function or wait till this PR got merged with that function ? 

I can make that before today afternoon Easily.

Comment (VipinDevelops) [reply src/components/Attachments/ImageAttachment.js:9]: hey @VenomFate-619  i made this function in my PR you can use it from there

File: src/components/Attachments/ImageAttachment.js (hunk 2 above)
Comment (VenomFate-619) [reply src/components/Attachments/ImageAttachment.js:11]: Yes, it will be.
But less than this width won't be there in the common case.

**Review by sidmohanty11 — APPROVED** [review]
Thanks @VenomFate-619 ! LGTM

**Pull request merged**